iio-widgets/src/iiowidgetfactory.cpp

```diff
@@ -195,6 +198,22 @@ IIOWidget *IIOWidgetFactory::buildSingle(uint32_t hint, IIOWidgetFactoryRecipe r
 		dataStrategy = new DeviceAttrDataStrategy(recipe, parent);
 	} else if(hint & ContextAttrData) {
 		dataStrategy = new ContextAttrDataStrategy(recipe, parent);
+	} else if(hint & CMDQAttrData) {
+		auto conn = recipe.connection;
+		if(conn) {
+			dataStrategy = new CmdQChannelAttrDataStrategy(recipe, conn->commandQueue(), parent);
+		} else {
```

Comment:
Do you still need to pass the command queue as a constructor param? After the last changes the recipe has a reference to the Connection obj, so it also has a reference to the command queue.